### DISCUSSION TITLE
Automated deletion of old chats in history

### DISCUSSION THREAD

**Discussion by yoranvanoirschot** [open-webui/open-webui#4580] [Ideas]
**Is your feature request related to a problem? Please describe.**
Indefinite storage of chats causes privacy concerns for companies. Employees may feed in PII in the chat and this is then stored indefinitely in the chat history. From the Europe GDPR this is not allowed, and people have the right to be forgotten. Also, in the long run performance issues could arise to the indefinitely increasing database size.

**Describe the solution you'd like**
We'd like to be able to configure the TTL of historic chats through an environment variable. A cronjob should then periodically delete any chats from the database which exceed this TTL.

**Describe alternatives you've considered**
An alternative would be to add an environment variable to disable the chat history for all users. This is however a bummer because most users feel it is extremely useful to get back to recent chats.

**Additional context**
-

**Comment by MNBussmann** [comment]
I'd welcome this feature as well. In addition to that, with the chats being deleted by either the TTL or manually by the user any uploads, audio files and image generations linked to that chat should be deleted as well, as currently they are persistent in the cache even after the chat has been deleted.

**Comment by KingstonJoseph12** [comment]
I'd welcome this feature as well. I am planning to deploy this UI for an organization. It will be helpful if I can set the time limit like for example, 30 days. Post then the chats and the uploaded documents in chat will auto deleted.

**Comment by m8dhouse** [comment]
Hi I made a small python script to remove chats older than x-days and also check for orphaned files. There is test switch where it does nothing - just log. You can run this as a cron job.

It checks the chats table and the json that holds  the chat details. If looks for files and then also updates the files table.
I've noticed there is a visualisations subfolder in the /uploads. For the moment it still sees that as an orphan. Needs some work in the code to ignore that (or actually check the chat details and then do the same scanning for this kind of files).

Please use with care. Haven't fully tested it yet with different situations.

Info in this discussion: https://github.com/open-webui/open-webui/discussions/5416

**Comment by Classic298** [comment]
Hello everyone.

I worked on a PR related to this discussion.
Testing wanted!

https://github.com/open-webui/open-webui/pull/16520
